packages/cache/src/internal/tar.ts

```diff
@@ -49,19 +54,28 @@ function getWorkingDirectory(): string {
 }
 
 // Common function for extractTar and listTar to get the compression method
-function getCompressionProgram(compressionMethod: CompressionMethod): string[] {
+async function getCompressionProgram(compressionMethod: CompressionMethod): Promise<string[]> {
   // -d: Decompress.
   // unzstd is equivalent to 'zstd -d'
   // --long=#: Enables long distance matching with # bits. Maximum is 30 (1GB) on 32-bit OS and 31 (2GB) on 64-bit.
   // Using 30 here because we also support 32-bit self-hosted runners.
+  const tarPath = await getTarPath([])
+  const BSD_TAR_ZSTD = IS_WINDOWS && tarPath === SystemTarPathOnWindows
   switch (compressionMethod) {
     case CompressionMethod.Zstd:
+      if (BSD_TAR_ZSTD) {
+        return ['-a'] // auto-detect compression
+      }
       return [
         '--use-compress-program',
-        IS_WINDOWS ? 'zstd -d --long=30' : 'unzstd --long=30'
+        IS_WINDOWS ? 'zstd -d --long=30' : 'unzstd --long=30',
+        '-tf'
```

Comment:
`-t` is used for?

Reply:
-t is used for listing the cache contents. I had to move it from the listTar function to getCompressionProgram due to ordering

Reply:
Okay. Makes sense